### DISCUSSION TITLE
Cross-app comments - in the board meeting

### DISCUSSION THREAD

**Discussion by jamescridland** [Podcastindex-org/podcast-namespace#423] [General]
A strange listen this week to the board meeting (Dec 23), as Adam says that nobody wants cross-app comments, and that it seems impossible to do.

```text
01:17:50,610 --> 01:17:54,570
I'm going to say it

01:17:54,570 --> 01:18:00,060
is dead. In the sense it is dead
in the sense that it is it will

01:18:00,060 --> 01:18:04,950
be clear at that point that
people do not want comments in

01:18:04,950 --> 01:18:05,700
their podcasts.

01:18:06,030 --> 01:18:09,090
I think that's clear already.
Honestly, I think it's clear.
```

I thought it might be interesting to précis where we are and posit some ideas on where we should be going.

## The user need

As a **developer**, I want to ensure that my podcast app offers features that bring people to my app and keep them using them. What will work for me best is features that are not available in Apple or Spotify - that will make listeners switch to my app because I support it and they don't. Ideally, I'd like to see features that offer me some kind of opportunity for revenue generation, too.

As a **podcaster**, I want to hear from my listeners. It's lonely making podcasts. Feedback helps me know that there are people out there listening to my work, and keeps me focused on them.
* At the time of writing, [Tom Scott's video about renewable energy](https://www.youtube.com/watch?v=nCrTsWtPVIY) has had 1.9m views and **3,063 comments**.
* Joe Rogan is apparently still complaining that there are no comments on his show on Spotify

As a **listener**, I want to contribute to the show with my thoughts and views. I'd like to do that in the place where I'm listening to the show - not
* There are 25,000 members for the [Darknet Diaries subreddit](https://www.reddit.com/r/darknetdiaries/).

As a **podcast host**, the biggest risk to my business is that people get bored of podcasting and stop producing a podcast. Anything that helps me retain a customer is an investment into my longevity as a podcast host. That could be encouraging my customers to keep podcasting; or be offering a product or service that my competitors do not produce.

The internet is littered with examples of comments - whether for YouTube videos, podcasts or news articles. I would suggest that it is evidence that **it is a feature that is wanted by all parts of the podcasting ecosystem**.

## Where we are

We have:

1. A tag called `socialInteract`. This proposal appears to require use of all of activitypub, twitter, and lightning, or alternatively requires use of one of these protocols (it's not clear from the specification whether you can "support" `socialInteract` if you only have support for one of these proposals. The tag also included other commenting systems, including Facebook and xmpp, but has been pruned back.
2. An entirely separate IRC channel for live podcasts where people can comment on the show they're currently listening to
3. "Boostagrams" - comments with an accompanying sat value, sent via lightning, which are private until sent, except if you use Fountain, where a boostagram is public and shown as a threaded comments window.

Three separate commenting systems - but wait, there's always time for a fourth!

```text
01:18:55,740 --> 01:18:58,320
I still think there's an interesting
place for noster.
```

But, as evidenced by the last twenty minutes of the board meeting, the last five minutes of Podnews Weekly Review, or the first five minutes of The New Media Show, **we already have a de-facto commenting system:** a boostagram, sent by listeners. These "comments" are sent directly to the podcaster, who reads them out in the next show (or, in Podnews Daily's case, never reads them out because it doesn't fit the output).

And, as Adam points out as soon as he says that nobody wants comments...

```text
01:18:09,810 --> 01:18:14,010
To me, the lightning
comments is one thing because

01:18:14,010 --> 01:18:17,670
it's booster grams, it's always
positive. There's no, I've not

01:18:17,670 --> 01:18:23,430
seen someone boost anything and
say this show sucks. Yeah. Or

01:18:23,430 --> 01:18:27,780
even even, I haven't even seen
anyone do spam. And you know,

01:18:27,780 --> 01:18:31,320
which will be interesting if you
pay to pay to play spam or ads

01:18:31,320 --> 01:18:37,410
or whatever.
```

So, as Adam says, we already *have* a commenting system which:
a) listeners are already using
b) is "always positive" for podcast hosts
c) that has spam filtering built-in (because who wants to pay to spam)
d) that puts money into the podcasting ecosystem.

A [proposal for it was put into the public Github in March](https://github.com/Podcastindex-org/podcast-namespace/issues/347) as to how this might formally work - a JSON end-point which pulls in the boosts for that episode, allowing any app to display them. 

There are issues - potential centralisation of this service is one of them; which could be avoided by either:
a) 1% split in the show to a commenting system would give that the detail needed to produce a JSON file of all the comments
b) an API from the creator's lightning wallet (like Alby) would enable lightning comments. Alby has an authenticated API that would do the trick; but it would ideally need to be unauthenticated, opt-in, and filtered. A third party could potentially do that work.

The discussion was warmly welcomed by some of us - but appeared to have stalled at the end of March. Neither Dave nor Adam have made any comments in the proposal.

If it's a bad idea, it would be great to have that discussion as to why, rather than pronouncing that cross-app comments are dead. I'm not sure they are.

(As a disclaimer: more than six months after Oscar's initial proposal, I became an advisor to Fountain; but we've not actually talked about this).

**Comment by MerryOscar** [comment]
My view is that any comments system for podcasting MUST include boosts.

However on reflection I believe the the spec for my original idea for Lightning Comments is not suitable and since March we have also updated the comments architecture within Fountain so it is outdated. I have therefore closed that issue.

My view is that [nostr](https://github.com/nostr-protocol/nostr) is actually very suitable for what we need and I would advocate going in this direction.

I plan to spend some time on this to figure out what it would look like and demonstrating it in Fountain.

**Reply by jamescridland** [reply]
Nostr certainly looks interesting, and has some implementation in Alby.

https://iris.to/#/profile/npub1gg3zv8wydfcrfytr9c3keyxhp4ju8ntgyqefcclfh3ztzflg0peqrlq4cg is, I think, me.

https://iris.to/#/profile/npub17pt6atea5u3ptqc6kj2zs2tl6lzpdynk0r0tvrj7fvppgk5ywfws3shfeq is, I think, Podnews (though you'd need to also add the relay wss://rsslay.nostr.net )

**Comment by samsethi** [comment]
Hi James

Podfans has implemented Boostergrams and integrated them with the Alby API.  So users can pay for their boosts but also we can pull in any other boostergrams from other apps that support WebLN which is acting as a central repository.  Tools like Saturn and this new Boostergram viewer https://boostagram-viewer-lnd.albylabs.com/ shows all the boostergrams.  Boosts may be a better way for all Podcast Apps to implement a comment system and share them, so all apps can see the same boosts.  

Today most podcast users are struggling to understand wallets, sats and boosts.  Having a secondary comments system via ActivityPub may cause further confusion.  Like Oscar mentioned, I am reading about NOSTR and NIP-05 and still haven't fully understood what they are and how they may be useful when applied to Podcasting.   

These blog posts from Alby might help.  
- https://blog.getalby.com/nostr-in-the-alby-extension/ 
- https://blog.getalby.com/cln-connector-announcement/

However this tweet today from Lyle Pratt from Vida piqued my interest.  https://twitter.com/lylepratt/status/1608308619031756803?s=20&t=zr0bQaZCPynePMgxn3k5qg

**- "now your @VIDAislive username is a working Nostr NIP-05 validation address, a lightning payment address, and also a working email address."**  

<img width="613" alt="SATurn" src="https://user-images.githubusercontent.com/3492066/210019888-e9ead4c1-4efa-411d-8f58-b1cab36c63a3.png">

**Reply by jamescridland** [reply]
Thanks, Sam. The Alby API works for the podcaster (if you sign in), but not for an app (which can’t sign in and look at each podcaster’s Alby stream). I’m keen for an experience that shows all comments to all users. Those who self-host their own nodes are left out of this, too.

I’ve played around with Nostr for a bit. It does seem quite well suited to comments, and because it’s tied into WebLN, could work nicely and fit alongside boostagrams. It does appear to rely on relays, which I guess potentially means that it isn’t quite as decentralised as you’d think it is. In particular, to follow Podnews on Nostr, you also need to know the relay that Podnews is on, and add it to your Nostr profile so you see those messages. Were Podcast Index to have it’s own relay, then that would be good, but also bad since it wouldn’t be a decentralised platform again. It’s all a bit difficult.

I really like coupling all comments with some form of payment. In Adam’s own words, it fixes issues with spam, negative comments, and flooding. I don’t think we need confuse users with wallets, sats and stuff - that’s marketing words. “Buy $5, get 500 comments that share your money with podcasters” works well for comments (though boostagrams are different of course).

**Comment by adamc199** [comment]
@samsethi The boostagram viewer *only* works with the getalby extension that is connected to a standalone node, not the stock getalby wallet.

@jamescridland I am staunchly against "by $5 get 500 comments" that is NOT value4value. It should remain open to the sender.

@samsethi & @jamescridland Please also consider appropriate disclosures if you are commenting on a service or product you are an advisor for.

**Reply by dhk2** [reply]
> @samsethi The boostagram viewer only works with the getalby extension that is connected to a standalone node, not the stock getalby wallet.

I have a basic getalby wallet and in my testing it works perfectly with SATurn and Conshax

**Comment by johnspurlock** [comment]
Hmm, I don't get the hair-on-fire when it comes to this topic.  We have a good spec, and quite fortunate that now we have hundreds of thousands of new users on that platform (ActivityPub), this is the right time to double down on that.

It seems critical to me that any successful open comment standard for podcasting not be gated by any particular crypto token, or a crypto token at all - they should all federate equally via AP.

Nothing is stopping servers and apps from integrating this today - just give it some time, it's not an easy spec and now there are hundreds of eyes on it - there will be other new apps and tools on the scene in the new year...

Happy new year!

**Reply by SpencerPearson** [reply]
If we're taking a temperature check here I have had no excitement or desire for having cross-app comments or comments in general in association with my own shows. We like boostagrams and they do everything we desire already. We still believe comments are among the lowest form of internet interaction and there's no reason to solicit feedback from our listeners other than in our IRC chat or via a boostagram with sats attached.

Again, I just speak for myself and my cohosts and listeners, but this has been our position for the past ~year and a half that cross-app comments have been in play. We don't have any interest in standing in the way of cross app comments but we have never made plans to adopt or promote them should they come to fruition.

**Reply by boobury** [reply]
This is incorrect, there's a lower tier. Leave a star please thank you

**Reply by theDanielJLewis** [reply]
I agree with @johnspurlock. It's not only the gating but also the high barrier to entry. Consider someone completely new to this:

> To comment "all" you have to do is get a Bitcoin or lighting wallet, buy some bitcoin, transfer that to your Lightning wallet if you bought the BTC on-chain, or transfer that to your podcast app, but you need a Podcasting 2.0 podcast app that supports V4V …

WAY too complicated! I'd love for commenting to be as simple as "enter name, email address, and comment" like it is for most website comments.

**Comment by genebean** [comment]
As both a listener who boosts most shows I consume and a podcaster, I absolutely want to see cross app comments. My ideal scenario is that it is centered on ActivityPub and that boostagrams are bridged to AP like Dave mentions [here](https://podcastindex.social/@dave/109609011137603532): 
![A18E43FC-279F-4634-82E9-B0E4A47707C5](https://user-images.githubusercontent.com/966789/210162258-df0c9955-259e-4009-bc01-fbde1265aa25.jpeg)

As @johnspurlock said, the ability to comment should not be predicated on paying money (or sats). 

Additionally, devs such as Marco from Overcast have made it clear they don’t want to bring boosts to their apps… tying cross app comments to crypto exclusively significantly limits the field of apps that would consider participating, and the number of people who’d potentially engage. If we want to build features Apple & Spotify don’t have, we should strive to have those features available in as many podcast apps as possible.

**Comment by theDanielJLewis** [comment]
I like the way that live chat works in YouTube or Twitch: regular chat messages are mixed in with "superchats" that include money. That's the kind of experience I'd love to see in cross-app comments, only time-shifted instead of live. (But live-refresh for comments would be great!)
